src/main/java/net/imglib2/roi/geom/integer/Line.java

```diff
@@ -0,0 +1,415 @@
+package net.imglib2.roi.geom.integer;
+
+import java.util.Iterator;
+
+import net.imglib2.Cursor;
+import net.imglib2.IterableInterval;
+import net.imglib2.Localizable;
+import net.imglib2.Point;
+import net.imglib2.Positionable;
+import net.imglib2.RealPoint;
+import net.imglib2.RealPositionable;
+import net.imglib2.Sampler;
+
+/**
+ * An immutable line sampled on integer position.
+ * <p>
+ * The cursors generated by the {@link IterableInterval} will iterate exactly
+ * once over all the integer locations on a line between in proper order from
+ * the specified start to the specified end points, included.
+ * <p>
+ * Two lines are considered equal iff their starting points and ending points
+ * are at to the same locations. This is reflected in the
+ * {@link #equals(Object)}, {@link #hashCode()} and {@link #iterationOrder()}
+ * methods.
+ * <p>
+ * This implementation uses floating-point logic instead of the pure integer
+ * logic of Bresenham line (<a href=
+ * "https://en.wikipedia.org/wiki/Bresenham%27s_line_algorithm">Wikipedia</a>)
+ * but the results are quasi identical and the performance penalty small.
+ * 
+ * @author Jean-Yves Tinevez
+ *
+ */
+public class Line implements IterableInterval< Void >
```

Comment:
I wouldn't call this class `Line` since there is already a `Line` in imglib2-roi. Maybe call this `IterableLine` or something else?

Also if you extend `AbstractInterval` you can remove a lot of the `Interval` boilerplate code.